lib/vehicle_factory.rb

```diff
@@ -0,0 +1,15 @@
+require './lib/vehicle'
+
+class VehicleFactory
+  def create_vehicles(data)
+    data.map do |car|
+      Vehicle.new({
+        vin: car[:vin_1_10],
+        year: car[:model_year].to_i,
+        make: car[:make],
+        model: car[:model],
+        engine: :ev
```

Comment:
The other thing to think about here, is a lot of the keys from the `car` hash you have in your block variable, already have the keys that your Vehicle class is looking for to initialize. You could just send in Vehicle.new(car) and then the Vehicle class would use what you have in initialize to pull out the info it needed. This wouldn't work for the vin since the car hash has the key as :vin_1_10, but you could add in the :vin key you needed:
car[:vin] = car[:vin_1_10] which would add that key with the correct value, and then you just send all that data, as is, to the Vehicle class. Hoping this all makes sense, reach out if it doesn't.